digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala

```diff
@@ -1210,6 +1247,37 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
     }
   }
 
+  private def getCachedRoadLinksWithComplementaryAndChanges(municipalityCode: Int): (Seq[RoadLink], Seq[ChangeInfo]) = {
+    val dir = getCacheDirectory
+    val cachedFiles = getCacheFiles(municipalityCode, dir, Some(true))
+    cachedFiles match {
+      case Some((geometryFile, changesFile)) =>
+        logger.info("Returning cached result")
+        (vvhSerializer.readCachedGeometry(geometryFile), vvhSerializer.readCachedChanges(changesFile))
+      case _ =>
+        val (roadLinks, changes) = reloadRoadLinksWithComplementaryAndChangesFromVVH(municipalityCode)
+        if (dir.nonEmpty) {
+          try {
+            val newGeomFile = new File(dir.get, geometryWithComplementaryCacheFileNames.format(municipalityCode, System.currentTimeMillis))
+            if (vvhSerializer.writeCache(newGeomFile, roadLinks)) {
+              logger.info("New cached file created: " + newGeomFile + " containing " + roadLinks.size + " items")
+            } else {
+              logger.error("Writing cached geom file failed!")
+            }
+            val newChangeFile = new File(dir.get, changeCacheFileNamesWithComplementary.format(municipalityCode, System.currentTimeMillis))

```

Comment:
The change info with or without complementary is the same, so we don't need to have a separated cache for it.